test/elixir/test/cookie_auth_test.exs

```diff
@@ -0,0 +1,415 @@
+defmodule CookieAuthTest do
+  use CouchTestCase
+
+  @moduletag :authentication
+
+  @users_db "_users"
+
+  @password "3.141592653589"
+
+  test "test cookie auth" do
+    # Create db if not exists
+    Couch.put("/#{@users_db}")
+
+    server_config = [
+      %{
+        :section => "chttpd_auth",
+        :key => "authentication_db",
+        :value => @users_db
+      },
+      %{
+        :section => "couch_httpd_auth",
+        :key => "authentication_db",
+        :value => @users_db
+      },
+      %{
+        :section => "couch_httpd_auth",
+        :key => "iterations",
+        :value => "1"
+      },
+      %{
+        :section => "admins",
+        :key => "jan",
+        :value => "apple"
+      }
+    ]
+
+    resp =
+      Couch.get(
+        "/#{@users_db}/_changes",
+        query: [feed: "longpoll", timeout: 5000, filter: "_design"]
+      )
+
+    assert resp.body
+
+    run_on_modified_server(server_config, &test_fun/0)
+  end
+
+  defp login(user, password) do
+    sess = Couch.login(user, password)
+    assert sess.cookie, "Login correct is expected"
+    sess
+  end
+
+  defp logout(session) do
+    assert Couch.Session.logout(session).body["ok"]
+  end
+
+  defp loginAsUser(user) do
+    pws = %{
+      "jan" => "apple",
+      "Jason Davies" => @password,
+      "jchris" => "funnybone"
+    }
+
+    user1 = Regex.replace(~r/[0-9]$/, user, "")
+    login(user1, pws[user])
+  end
+
+  defp create_doc_expect_error(db_name, doc, status_code, msg) do
+    resp = Couch.post("/#{db_name}", body: doc)
+    assert resp.status_code == status_code
+    assert resp.body["error"] == msg
+    resp
+  end
+
+  defp open_as(db_name, doc_id, options) do
+    use_session = Keyword.get(options, :use_session)
+    user = Keyword.get(options, :user)
+    expect_response = Keyword.get(options, :expect_response, 200)
+    expect_message = Keyword.get(options, :error_message)
+
+    session =
```

Comment:
There is a shortcut to express it:
```
session = Keyword.get(options, :use_session)
  || loginAsUser(user)
```

Reply:
or since you need value of `use_session` below. We can say: `session = use_session || loginAsUser(user)`